hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1119,14 +1183,14 @@ bool MDMParser::powerOff(void)
         }
         // Try to power off
         for (int i=0; i<3; i++) { // try 3 times
-            if (!_atOk()) {
+            if (_error || !_atOk()) {
                 if (_dev.dev == DEV_SARA_R410) {
                     check_ri = true;
                     // If memory issue is present, ensure we don't force a power off too soon
                     // to avoid hitting the 124 day memory housekeeping issue, AT+CPWROFF will
                     // handle this delay automatically, or timeout after 40s.
                     if (_memoryIssuePresent) {
```

Comment:
As long as we are going to be potentially entering this block more often, I think it would be safer to default to `_memoryIssuePresent = true` and set it to `false` only when `if (strcmp("L0.0.00.00.05.06,A.02.00", _verExtended) != 0) {`

Reply:
:+1: